File: dev-scripts/probe_api.py

import urllib.request
import json

def probe():
    print("Probing creator search API using Python urllib...")
    url = "https://go-api.accessprotocol.co/hub/search"
    data = {
        "sort": "featured",
        "page": 0,
        "per_page": 200
    }
    
    req = urllib.request.Request(url, data=json.dumps(data).encode('utf-8'), method='POST')
    req.add_header('Content-Type', 'application/json')
    
    try:
        with urllib.request.urlopen(req) as response:
            status = response.getcode()
            print(f"Status Code: {status}")
            body = response.read().decode('utf-8')
            parsed = json.loads(body)
            with open('creators_list.json', 'w') as f:
                json.dump(parsed, f, indent=2)
            print("Data saved to creators_list.json")
            if 'data' in parsed:
                print(f"Found {len(parsed['data'])} creators.")
    except Exception as e:
        print(f"Probe failed: {e}")

if __name__ == "__main__":
    probe()

File: dev-scripts/probe_api.js

const https = require('https');
const fs = require('fs');

async function probe() {
    console.log('Probing creator search API using built-in https module...');
    
    const data = JSON.stringify({
        sort: 'featured',
        page: 0,
        per_page: 100
    });

    const options = {
        hostname: 'go-api.accessprotocol.co',
        port: 443,
        path: '/hub/search',
        method: 'POST',
        headers: {
            'Content-Type': 'application/json',
            'Content-Length': data.length
        }
    };

    const req = https.request(options, (res) => {
        console.log(`Status Code: ${res.statusCode}`);
        let responseBody = '';

        res.on('data', (d) => {
            responseBody += d;
        });

        res.on('end', () => {
            console.log('Response received.');
            try {
                const parsed = JSON.parse(responseBody);
                fs.writeFileSync('creators_list.json', JSON.stringify(parsed, null, 2));
                console.log('Data saved to creators_list.json');
                if (parsed.data && Array.isArray(parsed.data)) {
                    console.log(`Found ${parsed.data.length} creators.`);
                }
            } catch (e) {
                console.log('Failed to parse JSON response.');
                console.log('Raw response:', responseBody.substring(0, 500));
            }
        });
    });

    req.on('error', (error) => {
        console.error('Request Error:', error);
    });

    req.write(data);
    req.end();
}

probe();
